mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/Transforms/BufferizableOpInterfaceImpl.cpp

```diff
@@ -153,18 +160,144 @@ struct EnqueueOpInterface
+    Location loc = op->getLoc();
+    rewriter.setInsertionPoint(enqueueAllocOp);
+
+    // Handle inputs
+    SmallVector<Value> newInputBuffers;
+    newInputBuffers.reserve(enqueueAllocOp.getInputs().size());
+    for (auto [idx, input] : llvm::enumerate(enqueueAllocOp.getInputs())) {
+      if (isScalar(&input)) {
+        newInputBuffers.push_back(input);
+        continue;
+      }
+      FailureOr<Value> buffer = getBuffer(rewriter, input, options);
+      if (failed(buffer))
+        return failure();
+
+      MemRefType memRefType = cast<MemRefType>(buffer->getType());
+
+      // Handle host tensor inputs
+      if (enqueueAllocOp.isOperandOnHost(idx) &&
+          !isInMemorySpace(memRefType, plan::MemorySpace::host_pinned)) {
+        FailureOr<Value> pinnedAlloc = options.createAlloc(
+            rewriter, loc,
+            MemRefType::get(memRefType.getShape(), memRefType.getElementType(),
+                            memRefType.getLayout(),
+                            plan::MemorySpaceAttr::get(
+                                ctx, plan::MemorySpace::host_pinned)),
+            ValueRange{});
+        if (failed(pinnedAlloc))
+          return failure();
+        if (failed(options.createMemCpy(rewriter, loc, *buffer, *pinnedAlloc)))
+          return failure();
+        newInputBuffers.push_back(*pinnedAlloc);
+        continue;
+      }
+
+      // Handle device tensor inputs
+      if (!enqueueAllocOp.isOperandOnHost(idx) &&
+          !isInMemorySpace(memRefType, plan::MemorySpace::device)) {
+        FailureOr<Value> devAlloc = options.createAlloc(
+            rewriter, loc,
+            MemRefType::get(
+                memRefType.getShape(), memRefType.getElementType(),
+                memRefType.getLayout(),
+                plan::MemorySpaceAttr::get(ctx, plan::MemorySpace::device)),
+            ValueRange{});
+        if (failed(devAlloc))
+          return failure();
+        if (failed(options.createMemCpy(rewriter, loc, *buffer, *devAlloc)))
+          return failure();
+        newInputBuffers.push_back(*devAlloc);
+        continue;
+      }
+
+      newInputBuffers.push_back(*buffer);
+    }
+
+    // Handle results
+    SmallVector<Type> outputBufferTypes;
+    outputBufferTypes.reserve(enqueueAllocOp.getNumResults());
+    for (unsigned i = 0; i < enqueueAllocOp.getNumResults(); ++i) {
+      Type resultType = enqueueAllocOp->getResultTypes()[i];
+      MemRefType memRefType;
+      if (auto tensorType = resultType.dyn_cast<TensorType>()) {
+        memRefType = MemRefType::get(tensorType.getShape(), tensorType.getElementType(),
+                                     MemRefLayoutAttrInterface(),
+                                     plan::MemorySpaceAttr::get(ctx, plan::MemorySpace::device));
+      } else {
+        memRefType = resultType.cast<MemRefType>();
+      }
+      outputBufferTypes.push_back(memRefType);
+    }
+
+    // Create the new operation
+    rewriter.create<EnqueueAllocOp>(
```

Comment:
You are missing some sort of API call to tell the bufferization driver that you are remapping the results of `op` to the results of this new op you are creating here.

Since you didn't do that, it i creating a `memref.alloc` to fill in the gap when it traverses later consumers.

I believe it should look like 

```
auto bufferizedOp = rewriter.create<EnqeueAllocOp>(...);
replaceOpWithBufferizedValues(rewriter, op, bufferizedOp.getResults());
```